src/generator.rs

```diff
@@ -96,9 +97,50 @@ pub fn run_generator(devices: &[Device], output_directory: &Path, fake_mode: boo
         }
     }
 
+    let compressors: BTreeSet<&String> = devices
+        .iter()
+        .filter(|device| device.compression_algorithm.is_some())
+        .flat_map(|device| device.compression_algorithm.as_ref())
```

Comment:
Having these two one after another is redundant: `Option::into_iter()` yields no elements for `None` and one element for `Some`, so filtering out `None` variants beforehand does nothing.